aiida/tools/importexport/dbimport/backends/django.py

```diff
@@ -0,0 +1,689 @@
+                raise exceptions.ImportValidationError(
+                    f"Node<{source.uuid}> already has an outgoing {link_type} link with label \"{link['label']}\""
+                )
+
+            # If the indegree is `unique` there cannot already be any other incoming links of that type,
+            # i.e., the target Node may not have a LinkType of current LinkType, coming in, existing already.
+            if indegree == 'unique' and (out_id, link['type']) in existing_incoming_unique:
+                raise exceptions.ImportValidationError(f'Node<{target.uuid}> already has an incoming {link_type} link')
+
+            # If the indegree is `unique_pair`,
+            # then the link labels for incoming links of this type should be unique,
+            # i.e., the target Node may not have a LinkType of current LinkType, coming in
+            # that also has the current Link label, existing already.
+            elif indegree == 'unique_pair' and \
+            (out_id, link['label'], link['type']) in existing_incoming_unique_pair:
+                raise exceptions.ImportValidationError(
+                    f"Node<{target.uuid}> already has an incoming {link_type} link with label \"{link['label']}\""
+                )
+
+            # New link
+            links_to_store.append(
+                models.DbLink(input_id=in_id, output_id=out_id, label=link['label'], type=link['type'])
+            )
+            if 'Link' not in ret_dict:
+                ret_dict['Link'] = {'new': []}
+            ret_dict['Link']['new'].append((in_id, out_id))
+
+            # Add new Link to sets of existing Links 'input PK', 'output PK', 'label', 'type'
+            existing_links.add((in_id, out_id, link['label'], link['type']))
+            existing_outgoing_unique.add((in_id, link['type']))
+            existing_outgoing_unique_pair.add((in_id, link['label'], link['type']))
+            existing_incoming_unique.add((out_id, link['type']))
+            existing_incoming_unique_pair.add((out_id, link['label'], link['type']))
+
+    # Store new links
+    if links_to_store:
+        IMPORT_LOGGER.debug('   (%d new links...)', len(links_to_store))
+
+        models.DbLink.objects.bulk_create(links_to_store, batch_size=batch_size)
+    else:
+        IMPORT_LOGGER.debug('   (0 new links...)')
+
+
+def _import_groups(
+    *, group_count: int, group_uuids: Iterable[Tuple[str, Set[str]]], foreign_ids_reverse_mappings: Dict[str, Dict[str,
+                                                                                                                   int]]
+):
+    """Import groups."""
+    from aiida.backends.djsite.db import models
+
+    if not group_count:
+        return
+
+    pbar_base_str = 'Groups - '
+
+    with get_progress_reporter()(total=group_count, desc=pbar_base_str) as progress:
+        for groupuuid, groupnodes in group_uuids:
+            if not groupnodes:
```

Comment:
I saw a similar logic elsewhere - it's not really clear to me why empty groups should be skipped?
In particular now that groups can store extras, one might well want to import an empty group

Reply:
This function does not actually store the groups (that is done in `_store_entity_data`), it just adds the nodes to the groups. I will change the name/docstring of the function to reflect that